networkx/generators/lattice.py

```diff
@@ -145,25 +145,30 @@ def grid_graph(dim, periodic=False):
 
 @nx._dispatchable(graphs=None, returns_graph=True)
 def hypercube_graph(n):
-    """Returns the *n*-dimensional hypercube graph.
-
-    The nodes are the integers between 0 and ``2 ** n - 1``, inclusive.
+    """
+    Return the n-dimensional hypercube graph.
 
-    For more information on the hypercube graph, see the Wikipedia
-    article `Hypercube graph`_.
 
-    .. _Hypercube graph: https://en.wikipedia.org/wiki/Hypercube_graph
+    The n-dimensional hypercube graph has 2^n nodes, each represented as an n-tuple
+    of binary values (0 or 1). Edges exist between nodes that differ in exactly one bit.
 
     Parameters
     ----------
     n : int
-        The dimension of the hypercube.
-        The number of nodes in the graph will be ``2 ** n``.
+        Dimension of the hypercube.
 
     Returns
     -------
-    NetworkX graph
-        The hypercube graph of dimension *n*.
+    Graph
+        The n-dimensional hypercube graph as an undirected graph.
+
+    Examples
+    --------
+    >>> G = nx.hypercube_graph(3)
+    >>> list(G.edges((0, 0, 0)))
+    [((0, 0, 0), (1, 0, 0)), ((0, 0, 0), (0, 1, 0)), ((0, 0, 0), (0, 0, 1))]
```

Comment:
```suggestion
    >>> list(G.neighbors((0, 0, 0)))
    [(1, 0, 0), (0, 1, 0), (0, 0, 1)]
```
I think this shows the neighbors of the node more cleanly?